management/server/idp/zitadel.go

```diff
@@ -561,12 +569,25 @@ func (zm *ZitadelManager) get(ctx context.Context, resource string, q url.Values
 			zm.appMetrics.IDPMetrics().CountRequestStatusError()
 		}
 
-		return nil, fmt.Errorf("unable to get %s, statusCode %d", reqURL, resp.StatusCode)
+		bodyBytes, _ := io.ReadAll(resp.Body)
+		zErr := zm.readZitadelError(bodyBytes)
+
+		return bodyBytes, fmt.Errorf("unable to get %s, statusCode %d, zitadel: %w", reqURL, resp.StatusCode, zErr)
 	}
 
 	return io.ReadAll(resp.Body)
 }
 
+func (zm *ZitadelManager) readZitadelError(errorBody []byte) error {
+	var zitadelErr zitadelErrorResponse
+	err := zm.helper.Unmarshal(errorBody, &zitadelErr)
+	if err != nil {
+		return fmt.Errorf("error unparsable body: %s", errorBody)
+	}
+
+	return fmt.Errorf("error code: %d message: %s", zitadelErr.Code, zitadelErr.Message)
+}
+
```

Comment:
We can pass the response body `io.ReadCloser` instead of body bytes `[]byte` to simplify  and reduce the need to handle reading the body in multiple places.

```go
func (zm *ZitadelManager) readZitadelError(body io.ReadCloser) error {
	bodyBytes, err := io.ReadAll(body)
	if err != nil {
		return fmt.Errorf("failed to read response body: %w", err)
	}


	var zitadelErr zitadelErrorResponse
	err = zm.helper.Unmarshal(bodyBytes, &zitadelErr)
	if err != nil {
		return fmt.Errorf("error unparsable body: %s", string(bodyBytes))
	}

	return fmt.Errorf("error code: %d message: %s", zitadelErr.Code, zitadelErr.Message)
}
```